scripts/checksum.sh

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+# checksum.sh
+#
+# SUMMARY
+#
+#   Creates a SHA256 checksum of all artifacts created during CI
+
+ROOT=$(git rev-parse --show-toplevel)
+VECTOR_VERSION=${VECTOR_VERSION:-nightly}
+
+pushd "${ROOT}/target/artifacts"
+
+shopt -s extglob
+ARTIFACTS=$(ls !(*SHA256SUMS))
```

Comment:
Given that you are using `bash`, you can use `mapfile` here to avoid the shellcheck warning later:
```bash
mapfile -t ARTIFACTS < <(ls !(*SHA256SUMS))
```
```bash
sha256sum "${ARTIFACTS[@]}" > vector-"$VECTOR_VERSION"-SHA256SUMS
```

Reply:
Neat, I made that change https://github.com/vectordotdev/vector/pull/18487

Thanks!